dance/transforms/filter.py

```diff
@@ -884,3 +885,83 @@ def gini_func(x, weights=None):
+                       ['min_counts', 'max_cells', 'min_cells', 'max_counts'],
+                       ['min_counts', 'max_cells', 'max_counts', 'min_cells'],
+                       ['min_cells', 'min_counts', 'max_counts', 'max_cells'],
+                       ['min_cells', 'min_counts', 'max_cells', 'max_counts'],
+                       ['min_cells', 'max_counts', 'min_counts', 'max_cells'],
+                       ['min_cells', 'max_counts', 'max_cells', 'min_counts'],
+                       ['min_cells', 'max_cells', 'min_counts', 'max_counts'],
+                       ['min_cells', 'max_cells', 'max_counts', 'min_counts'],
+                       ['max_counts', 'min_counts', 'min_cells', 'max_cells'],
+                       ['max_counts', 'min_counts', 'max_cells', 'min_cells'],
+                       ['max_counts', 'min_cells', 'min_counts', 'max_cells'],
+                       ['max_counts', 'min_cells', 'max_cells', 'min_counts'],
+                       ['max_counts', 'max_cells', 'min_counts', 'min_cells'],
+                       ['max_counts', 'max_cells', 'min_cells', 'min_counts'],
+                       ['max_cells', 'min_counts', 'min_cells', 'max_counts'],
+                       ['max_cells', 'min_counts', 'max_counts', 'min_cells'],
+                       ['max_cells', 'min_cells', 'min_counts', 'max_counts'],
+                       ['max_cells', 'min_cells', 'max_counts', 'min_counts'],
+                       ['max_cells', 'max_counts', 'min_counts', 'min_cells'],
+                       ['max_cells', 'max_counts', 'min_cells', 'min_counts']]
+
+
+@register_preprocessor("filter", "gene")
+class FilterGenesScanpyOrder(BaseTransform):
+    """Scanpy filtering gene transformation with additional options.
+
+    Parameters
+    ----------
+    order_index
+        Index of (min_counts,min_cells,max_counts,max_cells) order
+    min_counts
+        Minimum number of counts required for a gene to be kept.
+    min_cells
+        Minimum number (or ratio) of cells required for a gene to be kept.
+    max_counts
+        Maximum number of counts required for a gene to be kept.
+    max_cells
+        Maximum number (or ratio) of cells required for a gene to be kept.
+    split_name
+        Which split to be used for filtering.
+    channel
+        Channel to be used for filtering.
+    channel_type
+        Channel type to be used for filtering.
+
+    """
+
+    def __init__(self, order_index: int, min_counts: Optional[int] = None,
+                 min_cells: Optional[Union[float, int]] = None, max_counts: Optional[int] = None,
+                 max_cells: Optional[Union[float, int]] = None, split_name: Optional[str] = None,
+                 channel: Optional[str] = None, channel_type: Optional[str] = "X", **kwargs):
+        if order_index < 0 or order_index >= len(filter_genes_orders):
+            raise KeyError(f"An integer between 0 and {len(filter_genes_orders)-1} should be chosen")
```

Comment:
Is there any way a user can access the order info (index for the correct order they want) *without* looking into the source code, e.g., from the doc? If not, need to find alternative to make it more useable by users.

Reply:
The filter_genes_orders has been extracted in YML format, and I think that all conditional searches with discrete values can be adjusted to the search space of sequential parameters and stored in the conditional_parameter.yml